--- a/atl1e_main.c
+++ b/atl1e_main.c
@@ -24,6 +24,7 @@
  */
 
 
+#include "compat.h"
 #include "atl1e.h"
 
 char atl1e_driver_name[] = "ATL1e";
--- /dev/null
+++ b/compat.h
@@ -0,0 +1,15 @@
+/*
+ * Always define CONFIG_ATL1E_NAPI
+ * http://www.ussg.iu.edu/hypermail/linux/kernel/0806.3/1030.html
+ */
+#ifndef CONFIG_ATL1E_NAPI
+#define CONFIG_ATL1E_NAPI
+#endif
+
+/*
+ * Define required pci id
+ * http://www.ussg.iu.edu/hypermail/linux/kernel/0806.3/1021.html
+ */
+#ifndef PCI_DEVICE_ID_ATTANSIC_L1E
+#define PCI_DEVICE_ID_ATTANSIC_L1E 0x1026
+#endif
